packages/http-specs/specs/type/union/discriminated/main.tsp

```diff
@@ -0,0 +1,190 @@
+import "@typespec/http";
+import "@typespec/spector";
+
+using Http;
+using Spector;
+
+/**
+ * Describe scenarios for discriminated unions.
+ */
+@scenarioService("/type/union/discriminated")
+namespace Type.Union.Discriminated;
+
+// Models for discriminated unions
+model Cat {
+  name: string;
+  meow: boolean;
+}
+
+model Dog {
+  name: string;
+  bark: boolean;
+}
+
+/**
+ * Test discriminated union with default envelope serialization.
+ * The discriminated union should serialize with "kind" as discriminator
+ * and "value" as envelope property.
+ */
+@discriminated
+union PetWithEnvelope {
+  cat: Cat,
+  dog: Dog,
+}
+
+@route("/envelope")
+interface EnvelopeDiscriminated {
+  @scenario
+  @scenarioDoc("""
+    Test discriminated union with envelope serialization.
+    When query parameter "kind" is null or "cat", the expected response is:
```

Comment:
Is this the best practice to support null for query parameter then return the first kind in union as the default value?